crates/sargon/src/system/sargon_os/sargon_os_delete_account.rs

```diff
@@ -0,0 +1,260 @@
+use crate::prelude::*;
+use radix_engine_interface::blueprints::account::{
+    AccountRemoveAuthorizedDepositorInput as ScryptoAccountRemoveAuthorizedDepositorInput,
+    AccountRemoveResourcePreferenceInput as ScryptoAccountRemoveResourcePreferenceInput,
+};
+use std::future::Future;
+
+// ==================
+// Delete Account (Public)
+// ==================
+impl SargonOS {
+    pub async fn create_delete_account_manifest(
+        &self,
+        account_address: AccountAddress,
+    ) -> Result<TransactionManifest> {
+        let network_id = account_address.network_id();
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        let resource_preferences = gateway_client
+            .load_all_account_pages(account_address, |request| {
+                gateway_client.account_resource_preferences(request)
+            })
+            .await?
```

Comment:
I would opt for having a `fetchAllAccountResourcePreferences` function on the gateway_client, which is then used here. Same for `authorized_depositors`